x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java

```diff
@@ -2061,6 +2061,17 @@ public void testRateRequiresCounterTypes() {
         );
     }
 
+    public void testCoalesceWithMixedNumericTypes() {
```

Comment:
I guess this tests that the resulting plan passes the verifier? 
I guess that's OK, but wondering if maybe an alternative with a loop and verifying the type of the last argument wouldn't be more correct.